src/wizard/provideActionParameters/InputParameters.tsx

```diff
@@ -39,28 +39,45 @@ function InputParameters({
       initInputParametersData && initInputParametersData[item.name];
 
     const getFirstNotSetItemIdx = () => {
-      return inputParamsSchemas.findIndex((item, idx) => {
+      return inputParamsSchemas.findIndex((item) => {
         const initData =
           initInputParametersData && initInputParametersData[item.name];
         return !initData;
       });
     };
 
-    const onSuccessSubmit = (data: any) => {
+    const onSuccessSubmit = (data: string) => {
       setInputParameter(item.name, data);
+      if (inputParamsSchemas.length > 1) {
+        message.success(
+          `The '${item.name}' input parameters have been saved successfully`
+        );
+      }
       if (current + 1 >= inputParamsSchemas.length) {
-        setCurrent(getFirstNotSetItemIdx());
+        const idx = getFirstNotSetItemIdx();
+        if (idx === -1) {
+          // everything already set, do nothing
+          return;
+        }
+
+        setCurrent(idx);
       } else {
         setCurrent(current + 1);
       }
     };
 
+    const formButton: FormButton =
+      inputParamsSchemas.length === 1
+        ? { label: "Next", className: "form-submit-btn" }
+        : { label: "Save", className: "form-save-btn" };
+
```

Comment:
I need to say that two buttons distracts me a lot. Personally, I would vote to have the same experience (one btn) both when there is one input param and multiple params. 

In old approach we had:
![1 1](https://user-images.githubusercontent.com/17568639/149523417-e87e2181-a4bd-46ce-817c-7261c210b28d.png)

Currently, we have:
![1 2](https://user-images.githubusercontent.com/17568639/149523462-030eb439-f8d5-4e4a-8073-72973cc40ff7.png)


Suggested current approach:
![1 3](https://user-images.githubusercontent.com/17568639/149523530-3f1ab0ee-fca3-4b72-ac15-4e9203eceec0.png)

---- 

Additionally, I would add a simple feature—when we select a different tab, the content is also saved. IMO it's more convenient, as mostly people jump to other property but wants to save current work. Currently jumping between tabs is treated as "reset" btn (I know that it was like that previously, but with one button "next" I was less distracted, and I didn't run into that issue previously).

Reply:
> In old approach we had:

Remember that previously you weren't able to edit the parameters going back to this step. Also, navigating in tabs for a given step with "Next" button IMHO was really confusing for user. That's something I want to improve.

I agree with:

> Additionally, I would add a simple feature—when we select a different tab, the content is also saved. IMO it's more convenient, as mostly people jump to other property but wants to save current work.

Unfortunately I ran out of time to implement that properly - it would be difficult and time-consuming. I agree that the target approach would be to send form programmatically while switching tabs, and enable/disable Next button depending whether all forms are valid.

Right now I need to leave it as it is - I believe that this is a step in right direction in order to "untangle" and simplify the wizard flow.